Source/Details/ASDataController.h

```diff
@@ -115,6 +115,30 @@ AS_EXTERN NSString * const ASCollectionInvalidUpdateException;
 
 @end
 
+
+/**
+ * Delegate to describe if a parent is deallocating. Used
+ * in the batchUpdates script to check if the blocks should early
+ * return.
+ */
+
+@protocol ASDeallocateStatusProtocol <NSObject>
+
+/**
+ * Communicate to the child whether or not its parent is scheduled for deallocation.
+ *
+ * @discussion This is in response to a hard to reproduce crash where
+ * `performBatchUpdates` is sent on a valid pointer, however the internal UIKit
+ * collection view components such as _UICollectionViewData later reference a nil shared
+ * indexPath store during their many block invokes for a collection view update.
+ * This could be attributed the editingQueue still consuming its tasks
+ * while the collection view is being async deallocated. We want to early return
+ * at editing transation block invocation time since we can not destroy queued blocks;
+ */
+- (BOOL)isDeallocating;
```

Comment:
Should we merge this protocol with `ASDataControllerDelegate`? Most of the time data controller only cares about whether its delegate is still around or not.

Reply:
Yeah, ideally let's either merge it or generalize it by moving out to one of our utility-ish directories. I think it would also be appropriate to reference Texture PR #1290 for details and skip the full description, particularly because we haven't confirmed this theory yet (although it is very well-substantiated).